presto-main/src/main/java/com/facebook/presto/type/IntegerOperators.java

```diff
@@ -0,0 +1,198 @@
+import static com.facebook.presto.metadata.OperatorType.LESS_THAN_OR_EQUAL;
+import static com.facebook.presto.metadata.OperatorType.MODULUS;
+import static com.facebook.presto.metadata.OperatorType.MULTIPLY;
+import static com.facebook.presto.metadata.OperatorType.NEGATION;
+import static com.facebook.presto.metadata.OperatorType.NOT_EQUAL;
+import static com.facebook.presto.metadata.OperatorType.SUBTRACT;
+import static com.facebook.presto.spi.StandardErrorCode.DIVISION_BY_ZERO;
+import static com.facebook.presto.spi.StandardErrorCode.NUMERIC_VALUE_OUT_OF_RANGE;
+import static io.airlift.slice.Slices.utf8Slice;
+import static java.lang.String.valueOf;
+
+public final class IntegerOperators
+{
+    private IntegerOperators()
+    {
+    }
+
+    @ScalarOperator(ADD)
+    @SqlType(StandardTypes.INTEGER)
+    public static long add(@SqlType(StandardTypes.INTEGER) long left, @SqlType(StandardTypes.INTEGER) long right)
+    {
+        try {
+            return Math.addExact((int) left, (int) right);
+        }
+        catch (ArithmeticException e) {
+            throw new PrestoException(NUMERIC_VALUE_OUT_OF_RANGE, e);
+        }
+    }
+
+    @ScalarOperator(SUBTRACT)
+    @SqlType(StandardTypes.INTEGER)
+    public static long subtract(@SqlType(StandardTypes.INTEGER) long left, @SqlType(StandardTypes.INTEGER) long right)
+    {
+        try {
+            return Math.subtractExact((int) left, (int) right);
+        }
+        catch (ArithmeticException e) {
+            throw new PrestoException(NUMERIC_VALUE_OUT_OF_RANGE, e);
+        }
+    }
+
+    @ScalarOperator(MULTIPLY)
+    @SqlType(StandardTypes.INTEGER)
+    public static long multiply(@SqlType(StandardTypes.INTEGER) long left, @SqlType(StandardTypes.INTEGER) long right)
+    {
+        try {
+            return Math.multiplyExact((int) left, (int) right);
+        }
+        catch (ArithmeticException e) {
+            throw new PrestoException(NUMERIC_VALUE_OUT_OF_RANGE, e);
+        }
+    }
+
+    @ScalarOperator(DIVIDE)
+    @SqlType(StandardTypes.INTEGER)
+    public static long divide(@SqlType(StandardTypes.INTEGER) long left, @SqlType(StandardTypes.INTEGER) long right)
+    {
+        try {
+            return left / right;
+        }
+        catch (ArithmeticException e) {
+            throw new PrestoException(DIVISION_BY_ZERO, e);
+        }
+    }
+
+    @ScalarOperator(MODULUS)
+    @SqlType(StandardTypes.INTEGER)
+    public static long modulus(@SqlType(StandardTypes.INTEGER) long left, @SqlType(StandardTypes.INTEGER) long right)
+    {
+        try {
+            return left % right;
+        }
+        catch (ArithmeticException e) {
+            throw new PrestoException(DIVISION_BY_ZERO, e);
+        }
+    }
+
+    @ScalarOperator(NEGATION)
+    @SqlType(StandardTypes.INTEGER)
+    public static long negate(@SqlType(StandardTypes.INTEGER) long value)
+    {
+        return Math.negateExact((int) value);
```

Comment:
Shoot - sorry it looks like I messed up the commit order.. This is fixed in a later commit.